### PR TITLE
Added Mapbox token integration to Taipy

### PR DESCRIPTION
**Pull request by BaunaBadmosh**
This pull request updates the get_plotly_content function to use the get_mapbox_token function to retrieve the Mapbox token from the environment variable. The Mapbox token is then used to create a Plotly figure with a Mapbox map.
Issue #555 solved
The changes include:

Updated get_plotly_content function in taipy/gui/gui.py to use get_mapbox_token function
Added get_mapbox_token function to taipy/gui/utils/mapbox_token.py
Updated get_plotly_content function to use Mapbox token to create Plotly figure with Mapbox map
Note: The tests failed due to missing modules, which will need to be installed before the tests can be run successfully."
Thank You

### CODE REVIEW

File: taipy/gui/gui.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
 # specific language governing permissions and limitations under the License.
 
 from __future__ import annotations
-
+from taipy.gui.utils.mapbox_token import get_mapbox_token
```
Comment (FredLL-Avaiga) [inline taipy/gui/gui.py]: ruff will not agree with this

File: taipy/gui/gui.py [hunk 2]
```diff
@@ -445,6 +445,10 @@ def __process_content_provider(self, state: State, path: str, query: t.Dict[str,
                     if isinstance(content, PlotlyFigure):
 
                         def get_plotly_content(figure: PlotlyFigure):
+                            mapbox_token = get_mapbox_token()
```
Comment (FredLL-Avaiga) [inline taipy/gui/gui.py]: I think we wanted to support map token *not only* for figure but also for plotly.js based charts

File: taipy/gui/utils/mapbox_token.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import os
+
+def get_mapbox_token():
+    mapbox_token = os.environ.get('MAPBOX_TOKEN')
+    if mapbox_token is None:
+        raise ValueError("Mapbox Access token is not set as an environment variable")
```
Comment (FredLL-Avaiga) [inline taipy/gui/utils/mapbox_token.py]: this should stay optional

**Review by FredLL-Avaiga — CHANGES_REQUESTED** [review]

File: taipy/gui/gui.py [hunk 4]
```diff
@@ -452,42 +451,58 @@ def __process_content_provider(self, state: State, path: str, query: t.Dict[str,
...
+                                import json
+
+                                fig = go.Figure(data=content['data'], layout=content['layout'])
+                                fig.update_layout(mapbox_style="mapbox://styles/mapbox/streets-v11",
+                                mapbox_center_lat=37.7749, mapbox_center_lon=-122.4194,
+                                mapbox_zoom=12, mapbox_access_token=mapbox_token)
+                                html = fig.to_html(include_plotlyjs='cdn')
+                                return html
+
+                        Gui.register_content_provider((PlotlyFigure, dict), get_plotly_content)
```
Comment (FredLL-Avaiga) [inline taipy/gui/gui.py:475]: that won't work register_content_provider doesn't take a tuple as parameter

**Review by FredLL-Avaiga — CHANGES_REQUESTED** [review]

File: taipy/gui/gui.py (hunk 4 above)
Comment (BaunaBadmosh) [reply taipy/gui/gui.py:475]: sorry for making such mistakes. Shall I register the content provider for each type separately?

Comment (FredLL-Avaiga) [reply taipy/gui/gui.py:475]: I'm afraid it is more complicated than that
some Map chart don"t use figure at all and are completely built on the frontend
there are not using the content provider mechanism.

Comment (FredLL-Avaiga) [reply taipy/gui/gui.py:475]: don't apologize for making mistakes, that's the best way to learn :-)
Thanks for your help

Comment (BaunaBadmosh) [reply taipy/gui/gui.py:475]: thank you for your kind words. what shall I do ahead

Comment (FredLL-Avaiga) [reply taipy/gui/gui.py:475]: I propose that you understand how the chart component handle plotly.js charts on the frontend and figure a way to provide the right token from the backend to the frontend

Comment (BaunaBadmosh) [reply taipy/gui/gui.py:475]: okay sir i am on it.

**Pull request closed without merging**